utils/hooks/pre-commit

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Make sure we are inside the repository.
+cd "${BASH_SOURCE%/*}" &&
```

Comment:
Can you explain how this works?

Reply:
https://stackoverflow.com/questions/35006457/choosing-between-0-and-bash-source
https://gist.github.com/olegch/1730673

The `%/*` would strip the script name.

Not sure whether the logical `AND` is strictly required.